velox/exec/Aggregate.h

```diff
@@ -245,6 +245,11 @@ class Aggregate {
   // 'groups'. No-op for fixed length accumulators.
   virtual void destroy(folly::Range<char**> /*groups*/) {}
 
+  /// Returns true if current aggregate function is order sensitive.
+  virtual bool orderSensitive() const {
```

Comment:
It would be nice to make this metadata available in the registry without having to instantiate an instance of the function.

There is an example of this pattern in velox/expression/VectorFunction.h

```
struct VectorFunctionMetadata {
  /// Boolean indicating whether this function supports flattening, i.e.
  /// converting a set of nested calls into a single call.
  ///
  ///     f(a, f(b, f(c, d))) => f(a, b, c, d).
  ///
  /// For example, concat(string,...), concat(array,...), map_concat(map,...)
  /// Presto functions support flattening. Similarly, built-in special format
  /// AND and OR also support flattening.
  ///
  /// A function that supports flattening must have a signature with variadic
  /// arguments of the same type. The result type must be the same as input
  /// type.
  bool supportsFlattening{false};

  // TODO Add is-deterministic flag.
};
...
bool registerVectorFunction(
    const std::string& name,
    std::vector<FunctionSignaturePtr> signatures,
    std::unique_ptr<VectorFunction> func,
    VectorFunctionMetadata metadata = {},
    bool overwrite = true);
```

Perhaps, we can define AggregateFunctionMetadata struct with a single orderSensitive boolean and update registerAggregateFunction to take a parameter of type AggregateFunctionMetadata.

CC: @aditi-pandit @kagamiori

Reply:
Done.